agent/agent.go

```diff
@@ -461,15 +470,20 @@ func (a *agent) reportLifecycleLoop(ctx context.Context) {
 // setLifecycle sets the lifecycle state and notifies the lifecycle loop.
 // The state is only updated if it's a valid state transition.
 func (a *agent) setLifecycle(ctx context.Context, state codersdk.WorkspaceAgentLifecycle) {
+	report := agentsdk.PostLifecycleRequest{
+		State:     state,
+		ChangedAt: database.Now(),
```

Comment:
`database.Now()` or `time.Now()`?

Reply:
Intentionally used `database.Now()` here for consistency since it's a value that will be stored in the DB. Startup logs also use `database.Now()` but we do seem to have some mixed use in the agent and some other places may be wrong?

Thoughts @kylecarbs? For all intents and purposes, this shouldn't matter since the DB fields are `timestamptz`, what's the motivation for using `database.Now()` that always uses UTC? Logging purposes?

Reply:
> what's the motivation for using database.Now() that always uses UTC? Logging purposes?

+ It might be a good candidate for the linter rule.